packages/model-viewer/src/features/scene-graph/material.ts

```diff
@@ -28,6 +28,58 @@ const $pbrMetallicRoughness = Symbol('pbrMetallicRoughness');
 const $normalTexture = Symbol('normalTexture');
 const $occlusionTexture = Symbol('occlusionTexture');
 const $emissiveTexture = Symbol('emissiveTexture');
+export const $provideApplicator = Symbol('TextureApplicator');
+
+// Defines what a texture will be used for.
+export enum TextureUsage {
+  Base,
+  Metallic,
+  Normal,
+  Occlusion,
+  Emissive,
+}
+
+// Helper class for applying a texture to materials.
+export class TextureApplicator {
+  // Returns a lambda for applying a texture, with the specified state, to a
+  // material.
+  static[$provideApplicator](
+      onUpdate: () => void,
+      correlatedMaterials: Set<MeshStandardMaterial>|undefined,
+      usage: TextureUsage, encodding: TextureEncoding, format: PixelFormat) {
+    return (texture: ThreeTexture) => {
+      if (correlatedMaterials) {
+        for (const material of correlatedMaterials) {
+          texture.encoding = encodding;
+          texture.format = format;
+          texture.wrapS = RepeatWrapping;
+          texture.wrapT = RepeatWrapping;
```

Comment:
Okay, so this is our default values for a new texture. That's fine, but it seems odd that this creation all happens during `setTexture()`; shouldn't it be in `createTexture()` instead? Also, when creating a texture, you'll also need to create its children, like `sampler` which must have corresponding defaults. Is that all getting built consistently?

Reply:
Samplers are getting built - default samplers, I didn't have to change anything at least to make it work.
As for the other issue, these are not the default values these are values that get set/changed depending on the type of texture you want to set, thats why it's happing in setTexture. I can remove the wrap settings, I think those might be default values, left over from testing. Of-course we might want to provide some sort of override setting so you can specify your own settings, but these are following the glTF spec.

Reply:
Turns out clamp is the Three default, so I do need to keep repeat, repeat is the glTF default.

Reply:
moved setting the wrap and flipY to createTexture, and added a default encoding and format.